File: src/main/java/com/GoldenMine/parser/pronoun/PronounStorage.java
package com.GoldenMine.parser.pronoun;

import java.util.HashMap;

public class PronounStorage {
    public static final PronounStorage INSTANCE = new PronounStorage();

    private HashMap<String, IPronoun> pronouns = new HashMap<>();

    private PronounStorage() {
        init();
    }

    private void init() {
        addPronoun(new IPronoun것());
        addPronoun(new IPronoun나머지());
    }

    public IPronoun getPronoun(String key) {
        return pronouns.get(key);
    }

    public boolean hasPronoun(String key) {
        return pronouns.containsKey(key);
    }

    public void addPronoun(IPronoun pronoun) {
        pronouns.put(pronoun.getType(), pronoun);
    }
}

File: src/main/java/kr/goldenmine/parser/pronoun/PronounStorage.kt
package kr.goldenmine.parser.pronoun

import java.util.*

object PronounStorage {

    private val pronouns = HashMap<String, IPronoun>()

    init {
        init()
    }

    private fun init() {
        addPronoun(IPronoun것())
        addPronoun(IPronoun걸())
        addPronoun(IPronoun값())
        addPronoun(IPronoun나머지())
        addPronoun(IPronoun때())
    }

    fun getPronoun(key: String): IPronoun? {
        return pronouns[key]
    }

    fun isPronoun(key: String?): Boolean {
        return pronouns.containsKey(key)
    }

    fun addPronoun(pronoun: IPronoun) {
        pronouns[pronoun.type] = pronoun
    }

//    companion object {
//        val INSTANCE = PronounStorage()
//    }
}
